src/cli/download_tracker.rs

```diff
@@ -12,7 +12,7 @@ const DOWNLOAD_TRACK_COUNT: usize = 5;
 /// Tracks download progress and displays information about it to a terminal.
 pub struct DownloadTracker {
     /// Content-Length of the to-be downloaded object.
-    content_len: Option<u64>,
+    content_len: Option<usize>,
```

Comment:
This was mismatched with total_downloaded. Either we promote that to u64 or demote the length to usize, and since we're dealing with small archives, not disk images, usize seemed more appropriate. This existing mismatch was invisible to you because everything was being cast to f64, but if a web server did serve a 2.1G package, rustup would have failed when total_downloaded wrapped - after downloading 2G of data.
With this change, rustup will still fail, but when the content_len header is received. Which is much sooner. So I have shifted the detection earlier in the process but not introduced any new limitations to the code.